File: Exemplos M10/App_IntentDigitalHUB_Kotlin/Elgin_IntentDigitalHUB_Kotlin-font/app/src/main/java/com/elgin/intent_digital_hub/IntentDigitalHubService/SAT/Commands/EnviarDadosVenda.kt
package com.elgin.intent_digital_hub.IntentDigitalHubService.SAT.Commands

import com.elgin.intent_digital_hub.IntentDigitalHubService.SAT.SatCommand

class EnviarDadosVenda(
    private val numSessao: Int,
    private val codAtivacao: String,
    private val dadosVenda: String
) : SatCommand("EnviarDadosVenda") {
    override fun functionParameters(): String {
        return "\"numSessao\"" + ":" + numSessao + "," +
                "\"codAtivacao\"" + ":" + "\"" + codAtivacao + "\"" + "," +
                "\"dadosVenda\"" + ":" + "\"" + dadosVenda + "\""
    }
}

File: Exemplos M10/App_IntentDigitalHUB_Flutter/Elgin_IntentDigitalHUB_Flutter-font/lib/IntentDigitalHubService/SAT/enviar_dados_venda.dart
import 'package:intent_digital_hub/IntentDigitalHubService/SAT/sat_command.dart';

class EnviarDadosVenda extends SatCommand {
  final int _numSessao;
  final String _codAtivacao;
  final String _dadosVenda;

  EnviarDadosVenda(this._numSessao, this._codAtivacao, this._dadosVenda)
      : super('EnviarDadosVenda');

  @override
  functionParameters() {
    return '"numSessao"'
        ':'
        '$_numSessao'
        ','
        '"codAtivacao"'
        ':'
        '"$_codAtivacao"'
        ','
        '"dadosVenda"'
        ':'
        '"'
        '$_dadosVenda'
        '"';
  }
}
